src/Common/Core/Impl/Extensions/IOExtensions.cs

```diff
@@ -135,5 +135,15 @@ public static class IOExtensions {
             }
             return allowWrite && !denyWrite;
         }
+
+        public static string NormalizePath(this string path) {
```

Comment:
Does `GetFullPath` normalize? I thought it does nothing to rooted ie `c:\foo\.\..\` remains. But if it indeed does the right thing, we dont need `NormalizePath`.